packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -130,1076 +130,275 @@ typedef VMServiceConnector = Future<VMService> Function(Uri httpUri, {
+    // currently in the same way as hot reload, it leaves the tool free
+    // to change to a more efficient implementation in the future.
+    //
+    // `library` should be the file URI of the updated code.
+    // `class` should be the name of the Widget subclass to be marked dirty. For example,
+    // if the build method of a StatelessWidget is updated, this is the name of class.
+    // If the build method of a StatefulWidget is updated, then this is the name
+    // of the Widget class that created the State object.
+    vmService.registerServiceCallback('reloadMethod', (Map<String, dynamic> params) async {
+      final String libraryId = params['library'] as String;
+      final String classId = params['class'] as String;
+
+      if (libraryId.isEmpty) {
+        throw vm_service.RPCError(
+          "Invalid 'libraryId': $libraryId",
+          RPCErrorCodes.kInvalidParams,
+          '',
+        );
+      }
+      if (classId.isEmpty) {
+        throw vm_service.RPCError(
+          "Invalid 'classId': $classId",
+          RPCErrorCodes.kInvalidParams,
+          '',
+        );
+      }
 
-  /// Connect to a Dart VM Service at [httpUri].
-  ///
-  /// If the [reloadSources] parameter is not null, the 'reloadSources' service
-  /// will be registered. The VM Service Protocol allows clients to register
-  /// custom services that can be invoked by other clients through the service
-  /// protocol itself.
-  ///
-  /// See: https://github.com/dart-lang/sdk/commit/df8bf384eb815cf38450cb50a0f4b62230fba217
-  static Future<VMService> connect(
-    Uri httpUri, {
-      ReloadSources reloadSources,
-      Restart restart,
-      CompileExpression compileExpression,
-      ReloadMethod reloadMethod,
-      io.CompressionOptions compression = io.CompressionOptions.compressionDefault,
-      Device device,
-    }) async {
-    final VMServiceConnector connector = context.get<VMServiceConnector>() ?? VMService._connect;
-    return connector(httpUri,
-      reloadSources: reloadSources,
-      restart: restart,
-      compileExpression: compileExpression,
-      compression: compression,
-      device: device,
-      reloadMethod: reloadMethod,
-    );
-  }
+      globals.printTrace('reloadMethod not yet supported, falling back to hot reload');
 
-  static Future<VMService> _connect(
-    Uri httpUri, {
-    ReloadSources reloadSources,
-    Restart restart,
-    CompileExpression compileExpression,
-    ReloadMethod reloadMethod,
-    io.CompressionOptions compression = io.CompressionOptions.compressionDefault,
-    Device device,
-  }) async {
-    final Uri wsUri = httpUri.replace(scheme: 'ws', path: globals.fs.path.join(httpUri.path, 'ws'));
-    final io.WebSocket channel = await _openChannel(wsUri.toString(), compression: compression);
-    final StreamController<dynamic> primary = StreamController<dynamic>();
-    final StreamController<dynamic> secondary = StreamController<dynamic>();
```

Comment:
Not sure why the delegateService and service were using two streams [in the first place](https://github.com/flutter/flutter/pull/53809/files#diff-09557c5ab3307161425d441b6d004c20R35) but I trust you.

Reply:
They were single subscription, and I need to intercept the private streamNotify responses the old vm service was using to update Isolates